File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int nn=n;
        int sum=0;
        int p=1;
        while(n>0){
            int l=n%10;
            p*=l;
            sum+=l;
            n/=10;
        }
        return (nn%(p+sum))==0;
    }
};

File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        int sum=0;
        int pro=1;
        int tem=n;
        while(tem!=0){
           int rem=tem%10;
            sum+=rem;
            pro*=rem;
            tem/=10;
        }
        System.out.print(sum +" "+pro);
        int div=pro+sum;
        if(n%div==0)return true;
        return false;
    }
}